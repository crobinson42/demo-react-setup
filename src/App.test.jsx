import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import { add } from 'lodash'

it('Renders without blowing up', () => {
  const div = document.createElement('div');

  ReactDOM.render(<App />, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('Matches the snapshot', () => {
  const tree = renderer
    .create(<App />)
    .toJSON()

  expect(tree).toMatchSnapshot();
})

it('Knows the window location', () => {
  Object.defineProperty(window.location, 'href', {
    writable: true,
    value: 'http://test.com?param=123'
  });

  console.log(window.location.href)
})


it('To import a mock', () => {
  expect(add(1,2)).toEqual(3)
})
