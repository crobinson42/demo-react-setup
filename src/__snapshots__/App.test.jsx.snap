// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<div
  className="alert alert-info"
>
  <h3>
    Some other stuff
  </h3>
  <button
    className="btn btn-success"
  >
    Button
  </button>
  <div
    className="m-5"
  >
    <div
      className="card p-3 sscs-card"
    >
      Here is a BS4 card
    </div>
  </div>
</div>
`;
